Always fallback to real shell

pakrym-oai opened the pull request:
Either cmd.exe or `/bin/sh`.

pakrym-oai committed:
Always fallback to real shell

pakrym-oai committed:
Merge branch 'main' into pakrym/always-fallback-to-real-shell

# Conflicts:
#	codex-rs/core/src/shell.rs

pakrym-oai committed:
shell: fix default_user_shell fallback unwrap call

bolinfest reviewed (approved):
Is there an easy way to add a test to verify that if the user's default shell is fish that we end up choosing Bash (or Zsh on Mac)?

pakrym-oai committed:
tests: relax sh path and stabilize shell command checks

pakrym-oai committed:
shell: avoid String allocation for name and test fish fallback

Return &'static str from Shell::name to remove heap allocations.
Refactor default_user_shell via helper taking an optional path and
add macOS test ensuring fish defaults to zsh.

pakrym-oai committed:
bash: treat /bin/sh as valid shell for command extraction

jif-oai commented on codex-rs/core/src/shell.rs:
```diff
@@ -7,61 +7,41 @@ pub enum ShellType {
-            Shell::Unknown => None,
+    pub fn name(&self) -> &'static str {
+        match self.shell_type {
+            ShellType::Zsh => "zsh",
```
This will prevent custom shell definition if we don't use the shell path

pakrym-oai replied:
Yep, we can change it when we do.

jif-oai reviewed (approved):
A few comments but ok for me

pakrym-oai committed:
shell: use /bin/sh as ultimate fallback

Update ultimate_fallback_shell to prefer ShellType::Sh for /bin/sh and adjust
prompt caching tests to build expected environment context using Path::display()
and shell.name() directly.

pakrym-oai commented on codex-rs/core/src/shell.rs:
```diff
@@ -164,26 +164,56 @@ fn get_powershell_shell(path: Option<&PathBuf>) -> Option<PowerShellConfig> {
+        }
+    } else {
+        Shell {
+            shell_type: ShellType::Bash,
```
Good point, changed to sh!
